File: misago/users/serializers/ban.py
from django.utils.translation import ugettext as _

from rest_framework import serializers

from misago.core.utils import format_plaintext_for_html

from misago.users.models import Ban, BAN_IP


__all__ = ['BanMessageSerializer']


class BanMessageSerializer(serializers.ModelSerializer):
    message = serializers.SerializerMethodField()

    class Meta:
        model = Ban
        fields = ('message', 'expires_on')

    def get_message(self, obj):
        if obj.user_message:
            message = obj.user_message
        elif obj.check_type == BAN_IP:
            message = _("Your IP address is banned.")
        else:
            message = _("You are banned.")

        return {
            'plain': message,
            'html': format_plaintext_for_html(message)
        }

File: misago/frontend/misago/models/ban.js
(function (Misago) {
  'use strict';

  var Ban = function(data) {
    this.message = {
      html: data.message.html,
      plain: data.message.plain,
    };

    this.expires_on = data.expires_on;
  };

  var deserializeBan = function(data) {
    data.expires_on = Misago.deserializeDatetime(data.expires_on);

    return data;
  };

  Misago.addService('model:ban', function(_) {
    _.models.add('ban', {
      class: Ban,
      deserialize: deserializeBan
    });
  },
  {
    after: 'models'
  });
} (Misago.prototype));
